lib/AST/Stmt.cpp

```diff
@@ -321,9 +326,18 @@ SourceLoc StmtConditionElement::getStartLoc() const {
   case StmtConditionElement::CK_Availability:
     return getAvailability()->getStartLoc();
   case StmtConditionElement::CK_PatternBinding:
+    SourceLoc Start;
     if (IntroducerLoc.isValid())
-      return IntroducerLoc;
-    return getPattern()->getStartLoc();
+      Start = IntroducerLoc;
+    else
+      Start = getPattern()->getStartLoc();
+    
+    SourceLoc End = getInitializer()->getEndLoc();
```

Comment:
I'm really not happy with this duplicated logic. At this point it probably makes sense to just not implement `getStartLoc` and `getEndLoc`, right?